src/resources/cpp/language-basics.md

```diff
@@ -1,15 +1,307 @@
+These are commonly used to just leave quick thoughts in your code.
+
+```cpp
+/*
+ * This is a multiline comment
+ */
+```
+
+Multiline comments start with a `/*` and last until a `*/`. These are commonly used to leave longer thoughts in your
+code, such as detailed documentation. You may frequently see many developers use `//` on multiple lines in a row instead
+of using the dedicated multiline comment style.
 
 ## Keywords
 
-## Initialization vs Assignment
+C++ reserves a set of words in the language for its own use. Developers may use these keywords, but meaning cannot be
+provided to these keywords. In the above example on [Expression Statements](#expression-statements), `int` is a keyword.
+You can find a list of all keywords on [cppreference](https://en.cppreference.com/w/cpp/keyword). Fundamental types are
+an example of keywords. These types will be covered in more detail [later](#fundamental-data-types-and-literals).
+
+In addition to keywords, C++ has a concept of identifiers with special meanings. These are frequently available for use,
+but are reserved for the language to use in certain contexts. We will cover these in more detail in later chapters.
+
+```cpp
+int my_integer_value = 0; // int is a keyword
+float pi = 3.14f;         // float is a keyword
+```
+
+## Initialization Versus Assignment
+
+When we first create a variable in C++, it undergoes a process called initialization. Initialization is the process of
+providing a variable with an initial value.
+
+```cpp
+int foo = 0;
+```
+
+The above statement is one of the most common ways to initialize a variable. It gives the variable `foo` an initial
+value of `0`. There are many ways to initialize a variable, each having its own pros and cons. The different types of
+initialization will be discussed in more detail in another section.
+
+Assignment is a process that takes place after a variable already exists. It is the process of assigning a new value to
+an existing variable.
+
+```cpp
+int foo = 0; // Initialization
+foo = 2;     // Assignment
+```
+
+Like initialization, assignment has several different forms that will be discussed in more dtail in a later chapter.
 
 ## Fundamental Data Types and Literals
 
+Data types are a way in code to represent a unit of data. Programmers will build data types using other data types. C++
+provides developers with a set of data types built into the language, called fundamental types. Fundamental types cannot
+be split further without losing meaning. You can think of these types as the "atoms" of your programs.
+
+### Booleans
+
+The most fundamental type that can hold a value is a boolean. A boolean is a representation of `true` or `false`.
+
+### Integer Types
+
+An integer type in C++ is the same as the integer concept in math. Integer types represent real numbers with no decimal
+part. C++ provides integer types that can support different ranges of values.
```

Comment:
I don't like the phrase "integer concept in math". Saying that it's *the same* is also wrong and immediately corrected by the fact that you define value ranges, which mathematical integers don't have.

Note that you literally call them *integers* in math. If a reader doesn't what what an integer in math is, they're also not going to know what a real number in math is, so this semi-definition adds no value.

```suggestion
Integer types in C++ are like mathematical integers, except that they have a limited range of values.